### ISSUE TITLE
setState warning if using react router 4 and reduxForm

### ISSUE THREAD

**Issue by mitjade** [erikras/redux-form#2743]
### Are you submitting a **bug report** or a **feature request**?
bug

### What is the current behavior?
If I wrap a react component with `reduxForm` when I navigate to a different page via `react router` 4, I get a `setState` Warning.

Form otherwise works.

### What is the expected behavior?
If I use `react router` 3, this warning is not present.

### What's your environment?
`react-router` 4.0.0, `redux-form` 6.6.0

### Other informations
`Warning: setState(...): Cannot update during an existing state transition (such as within render or another component's constructor). Render methods should be a pure function of props and state; constructor side-effects are an anti-pattern, but can be moved to componentWillMount.`

- Stack trace:
`printWarning @ vendor.5d2ae58…_dll.js:53440`
`warning @ vendor.5d2ae58…_dll.js:53464`
`getInternalInstanceReadyForUpdate @ vendor.5d2ae58…_dll.js:191563`
`enqueueSetState @ vendor.5d2ae58…_dll.js:191709`
`vendor../node_modules/react/lib/ReactComponent.js.ReactComponent.setState @ vendor.5d2ae58…_dll.js:201090`
`onStateChange @ vendor.5d2ae58…_dll.js:196713`
`dispatch @ vendor.5d2ae58…_dll.js:217730`
`dispatch @ VM115946:2`
`(anonymous) @ vendor.5d2ae58…_dll.js:198093`
`(anonymous) @ vendor.5d2ae58…_dll.js:6825`
`dispatch @ VM115946:2`
`render @ vendor.5d2ae58…_dll.js:197969`
`render @ vendor.5d2ae58…_dll.js:198489`
`(anonymous) @ vendor.5d2ae58…_dll.js:183933`
`measureLifeCyclePerf @ vendor.5d2ae58…_dll.js:183212`
`_renderValidatedComponentWithoutOwnerOrContext @ vendor.5d2ae58…_dll.js:183932`
`_renderValidatedComponent @ vendor.5d2ae58…_dll.js:183959`
`_updateRenderedComponent @ vendor.5d2ae58…_dll.js:183883`
`_performComponentUpdate @ vendor.5d2ae58…_dll.js:183861`
`updateComponent @ vendor.5d2ae58…_dll.js:183782`
`receiveComponent @ vendor.5d2ae58…_dll.js:183684`
`receiveComponent @ vendor.5d2ae58…_dll.js:190560`
`_updateRenderedComponent @ vendor.5d2ae58…_dll.js:183891`
`_performComponentUpdate @ vendor.5d2ae58…_dll.js:183861`
`updateComponent @ vendor.5d2ae58…_dll.js:183782`
`performUpdateIfNecessary @ vendor.5d2ae58…_dll.js:183698`
`performUpdateIfNecessary @ vendor.5d2ae58…_dll.js:190592`
`runBatchedUpdates @ vendor.5d2ae58…_dll.js:191892`
`perform @ vendor.5d2ae58…_dll.js:193822`
`perform @ vendor.5d2ae58…_dll.js:193822`
`perform @ vendor.5d2ae58…_dll.js:191831`
`flushBatchedUpdates @ vendor.5d2ae58…_dll.js:191914`
`closeAll @ vendor.5d2ae58…_dll.js:193888`
`perform @ vendor.5d2ae58…_dll.js:193835`
`batchedUpdates @ vendor.5d2ae58…_dll.js:187589`
`batchedUpdates @ vendor.5d2ae58…_dll.js:191839`
`dispatchEvent @ vendor.5d2ae58…_dll.js:188025`

**Comment by mitjade** [comment]
Might be due to [this](https://github.com/ReactTraining/react-router/issues/4713)

**Comment by mitjade** [comment]
`react-router-redux 5.0.0-alpha.3` was the problem, fixed with `react-router-redux 5.0.0-alpha.4`

**Comment by rygine** [comment]
@mitjade I'm not using `ConnectedRouter` and I get this warning after a redirect to my login page that uses `redux-form`. Before the redirection occurs, I dispatch an action that resets my state. Not sure if that has anything to do with it.

Here's the route that triggers the warning:

```javascript
<Route path='/logout', render={() => {
  store.dispatch(Actions.logout());
  return <Redirect to='/login' />;
}}>
```

If you'd like, I can dig deeper.
